components/action-mgt/org.wso2.carbon.identity.action.management/src/test/java/org/wso2/carbon/identity/action/management/dao/ActionManagementDAOImplTest.java

```diff
@@ -0,0 +1,341 @@
+                        .build())
+                .build();
+        Action result = daoImpl.updateActionEndpoint(action.getType().getActionType(),action.getId(), endpointConfig,action.getEndpoint()
+                .getAuthentication(), TENANT_ID);
+        Assert.assertNotEquals(action.getEndpoint().getUri(),result.getEndpoint().getUri());
+        Assert.assertEquals(AuthType.AuthenticationType.BEARER, result.getEndpoint().getAuthentication().getType());
+    }
+
+    private Action getAction(String id){
+        return new Action.ActionResponseBuilder()
+                .id(id)
+                .type(Action.ActionTypes.PRE_ISSUE_ACCESS_TOKEN)
+                .name("Pre Issue Access Token")
+                .description("To configure pre issue access token")
+                .status(Action.Status.ACTIVE)
+                .endpoint( new EndpointConfig.EndpointConfigBuilder()
+                        .uri("https://example.com")
+                        .authentication(new AuthType.AuthTypeBuilder()
+                                .type(AuthType.AuthenticationType.BASIC)
+                                .properties(Arrays.asList(
+                                        new AuthProperty.AuthPropertyBuilder().name("username").value("admin").isConfidential(true).build(),
+                                        new AuthProperty.AuthPropertyBuilder().name("password").value("admin").isConfidential(true).build()))
+                                .build())
+                        .build())
+                .build();
+    }
+
+    private Action getActionWithoutOptionals(String id){
+        return new Action.ActionResponseBuilder()
+                .id(id)
+                .type(Action.ActionTypes.PRE_ISSUE_ACCESS_TOKEN)
+                .name("Pre Issue Access Token")
+                .status(Action.Status.ACTIVE)
+                .endpoint( new EndpointConfig.EndpointConfigBuilder()
+                        .uri("https://example.com")
+                        .authentication(new AuthType.AuthTypeBuilder()
+                                .type(AuthType.AuthenticationType.BASIC)
+                                .properties(Arrays.asList(
+                                        new AuthProperty.AuthPropertyBuilder().name("username").value("admin").isConfidential(true).build(),
+                                        new AuthProperty.AuthPropertyBuilder().name("password").value("admin").isConfidential(true).build()))
+                                .build())
+                        .build())
+                .build();
+    }
+
+    private void loadMockDatabaseUtil() {
+        try {
+            identityDatabaseUtil.when(() -> IdentityDatabaseUtil.getDBConnection(anyBoolean()))
+                    .thenReturn(getConnection());
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static Connection getConnection() throws SQLException {
```

Comment:
If we don't need to mock getDBConnection() in the before method class, remove this method and move the content to `loadMockDatabaseUtil()` method